tests/README.md

```diff
@@ -146,7 +146,111 @@ with the `zfs-tests.sh` wrapper script will look something like this:
     Running Time:	02:35:33
     Percent passed:	95.6%
     Log directory:	/var/tmp/test_results/20180515T054509
+
+### 4) Example of adding and running test-case (zpool_example)
+
+  This broadly boils down to 5 steps
+  1. Create/Set password-less sudo for user running test case.
+  2. Edit configure.ac, Makefile.am appropriately
+  3. Create/Modify .run files
+  4. Create actual test-scripts
+  5. Run Test case
+
+  Will look at each of them in depth.
+
+  * Set password-less sudo for 'Test' user as test script cannot be run as root
+  * Edit file **configure.ac** and include line under AC_CONFIG_FILES section
+    ~~~~
+      tests/zfs-tests/tests/functional/cli_root/zpool_example/Makefile
+    ~~~~
+  * Edit file **tests/runfiles/Makefile.am** and add line *zpool_example*.
+    ~~~~
+      pkgdatadir = $(datadir)/@PACKAGE@/runfiles
+      dist_pkgdata_DATA = \
+        zpool_example.run \
+        common.run \
+        freebsd.run \
+        linux.run \
+        longevity.run \
+        perf-regression.run \
+        sanity.run \
+        sunos.run
+    ~~~~
+  * Create file **tests/runfiles/zpool_example.run**. This defines the most
+    common properties when run with test-runner.py or zfs-test.sh.
+    ~~~~
+      [DEFAULT]
+      timeout = 600
+      outputdir = /var/tmp/test_results
+      tags = ['functional']
+
+      tests = ['zpool_example_001_pos']
+    ~~~~
+    If adding test-case to an already existing suite the runfile would
+    already be present and it needs to be only updated. For example, adding
+    **zpool_example_002_pos** to the above runfile only update the **"test ="**
+    section of the runfile as shown below
+    ~~~~
+      [DEFAULT]
+      timeout = 600
+      outputdir = /var/tmp/test_results
+      tags = ['functional']
+
+      tests = ['zpool_example_001_pos', 'zpool_example_002_pos']
+    ~~~~
+
+  * Edit **tests/zfs-tests/tests/functional/cli_root/Makefile.am** and add line
+    under SUBDIRS.
+    ~~~~
+      zpool_example \ (Make sure to escape the line end as there will be other folders names following)
+    ~~~~
+  * Create new file **tests/zfs-tests/tests/functional/cli_root/zpool_example/Makefile.am**
+    the contents of the file could be as below. What it says it that now we have
+    a test case *zpool_example_001_pos.ksh*
+    ~~~~
+      pkgdatadir = $(datadir)/@PACKAGE@/zfs-tests/tests/functional/cli_root/zpool_example
+      dist_pkgdata_SCRIPTS = \
+        zpool_example_001_pos.ksh
+    ~~~~
+  * We can now create our test-case zpool_example_001_pos.ksh under
+    **tests/zfs-tests/tests/functional/cli_root/zpool_example/**.
+    ~~~~
+	# DESCRIPTION:
+	#	zpool_example Test
+	#
+	# STRATEGY:
+	#	1. Demo a very basic test case
+	#
+
+	DISKS_DEV1="/dev/loop0"
+	DISKS_DEV2="/dev/loop1"
+	TESTPOOL=EXAMPLE_POOL
+
+	function cleanup
+	{
+		# Cleanup
+		destroy_pool $TESTPOOL
+		log_must rm -f $DISKS_DEV1
+		log_must rm -f $DISKS_DEV2
+	}
+
+	# Run function "cleanup" on exit
+	log_onexit cleanup
+
```

Comment:
Test cases should also `log_assert "Something."` somewhere in the opening.

Reply:
I am not sure if this is required for the given sample.  However, I will go through the example and incorporate your suggestion.

Reply:
Done. Thanks! I see what you meant +1: